examples/arkode/C_serial/ark_kepler.c

```diff
@@ -0,0 +1,812 @@
+/* clang-format off */
+/* ----------------------------------------------------------------------------
+ * Programmer(s): Cody J. Balos @ LLNL
+ * ----------------------------------------------------------------------------
+ * SUNDIALS Copyright Start
+ * Copyright (c) 2002-2022, Lawrence Livermore National Security
+ * and Southern Methodist University.
+ * All rights reserved.
+ *
+ * See the top-level LICENSE and NOTICE files for details.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ * SUNDIALS Copyright End
+ * ----------------------------------------------------------------------------
+ * We consider the Kepler problem. We choose one body to be the center of our
+ * coordinate system and then we use the coordinates q = (q1, q2) to represent
+ * the position of the second body relative to the first (center). This yields
+ * the ODE:
+ *    dq/dt = [ p1 ]
+ *            [ p2 ]
+ *    dp/dt = [ -q1 / (q1^2 + q2^2)^(3/2) ]
+ *          = [ -q2 / (q1^2 + q2^2)^(3/2) ]
+ * with the initial conditions
+ *    q(0) = [ 1 - e ],  p(0) = [        0          ]
+ *           [   0   ]          [ sqrt((1+e)/(1-e)) ]
+ * where e = 0.6 is the eccentricity.
+ *
+ * The Hamiltonian for the system,
+ *    H(p,q) = 1/2 * (p1^2 + p2^2) - 1/sqrt(q1^2 + q2^2)
+ * is conserved as well as the angular momentum,
+ *    L(p,q) = q1*p2 - q2*p1.
+ *
+ * By default we solve the problem by letting y = [ q, p ]^T then using a 4th
+ * order symplectic integrator via the SPRKStep time-stepper of ARKODE with a
+ * fixed time-step size.
+ *
+ * The rootfinding feature of SPRKStep is used to count the number of complete orbits. 
+ * This is done by defining the function,
+ *    g(q) = q2
+ * and providing it to SPRKStep as the function to find the roots for.
+ *
+ * The program also accepts command line arguments to change the method
+ * used and time-stepping strategy. The program has the following CLI arguments:
+ * 
+ *   --step-mode <fixed, adapt>  should we use a fixed time-step or adaptive time-step (default fixed)
```

Comment:
Remove for now?

Reply:
Still relevant for ARKStep which is an example option